packages/hoppscotch-app/components/realtime/Subscription.vue

```diff
@@ -0,0 +1,144 @@
+            {{ t("mqtt.qos") }}
+          </label>
+          <tippy
+            ref="QoSOptions"
+            interactive
+            trigger="click"
+            theme="popover"
+            arrow
+          >
+            <template #trigger>
+              <span class="select-wrapper">
+                <ButtonSecondary
+                  class="pr-8 ml-2 rounded-none"
+                  :label="`${QoS}`"
+                />
+              </span>
+            </template>
+            <div class="flex flex-col" role="menu">
+              <SmartItem
+                v-for="item in QoSValues"
+                :key="`qos-${item}`"
+                :label="`${item}`"
+                :icon="
+                  QoS === item
+                    ? 'radio_button_checked'
+                    : 'radio_button_unchecked'
+                "
+                :active="QoS === item"
+                @click.native="
+                  () => {
+                    QoS = item
+                    QoSOptions.tippy().hide()
+                  }
+                "
+              />
+            </div>
+          </tippy>
+        </div>
+
+        <div class="flex items-center">
+          <label
+            for="select-color"
+            class="px-4 font-semibold text-secondaryLight"
+          >
+            {{ t("mqtt.color") }}
+          </label>
+          <input
+            id="select-color"
+            v-model="color"
+            type="color"
+            class="px-1 rounded-md"
+          />
+        </div>
+      </div>
+
+      <div class="flex flex-col px-2">
+        <input
+          id="selectLabelAdd"
+          v-model="name"
+          v-focus
+          class="input floating-input"
+          placeholder=" "
+          type="text"
+          autocomplete="off"
+          @keyup.enter="addNewSubscription"
+        />
+        <label for="selectLabelAdd">
+          {{ t("action.label") }}
+        </label>
+      </div>
+    </template>
+    <template #footer>
+      <span>
+        <ButtonPrimary
+          :label="t('mqtt.subscribe')"
+          :loading="loadingState"
+          @click.native="addNewSubscription"
+        />
+        <ButtonSecondary
+          :label="t('action.cancel')"
+          @click.native="hideModal"
+        />
+      </span>
+    </template>
+  </SmartModal>
+</template>
+
+<script lang="ts" setup>
+import { ref } from "@nuxtjs/composition-api"
+import { MQTTTopic } from "~/helpers/realtime/MQTTConnection"
+import { useI18n, useToast } from "~/helpers/utils/composables"
+
+const toastr = useToast()
+const t = useI18n()
+
+const QoSOptions = ref<any>()
+
+defineProps({
+  show: {
+    type: Boolean,
+    default: false,
+  },
+  loadingState: {
+    type: Boolean,
+    default: false,
+  },
+})
+
+const emit = defineEmits<{
+  (e: "hide-modal"): void
+  (e: "submit", body: MQTTTopic): void
+}>()
+
+const QoSValues = [2, 1, 0] as const
+const QoS = ref<2 | 1 | 0>(2)
```

Comment:
Compile time constants should be UPPER_SNAKECASE.

Also, it is better to bind the types of these together.

```suggestion
const QoS_VALUES = [2, 1, 0] as const
const QoS = ref<typeof QoS_VALUES[number]>(2)
```